sdk/servicebus/Microsoft.Azure.WebJobs.Extensions.ServiceBus/src/Config/ServiceBusOptions.cs

```diff
@@ -46,83 +29,149 @@ public ServiceBusOptions()
-            if (SessionHandlerOptions != null)
+            set
             {
-                sessionHandlerOptions = new JObject
-                {
-                    { nameof(SessionHandlerOptions.AutoComplete), SessionHandlerOptions.AutoComplete },
-                    { nameof(SessionHandlerOptions.MaxAutoRenewDuration), SessionHandlerOptions.MaxAutoRenewDuration },
-                    { nameof(SessionHandlerOptions.MaxConcurrentSessions), SessionHandlerOptions.MaxConcurrentSessions },
-                    { nameof(SessionHandlerOptions.MessageWaitTimeout), SessionHandlerOptions.MessageWaitTimeout }
-                };
+                Argument.AssertAtLeast(value, 1, nameof(MaxConcurrentCalls));
+                _maxConcurrentCalls = value;
             }
+        }
+        private int _maxConcurrentCalls = Utility.GetProcessorCount() * 16;
 
-            JObject batchOptions = null;
-            if (BatchOptions != null)
+        /// <summary>
+        /// Gets or sets the maximum number of sessions that can be processed concurrently by a function.
+        /// The default value is 8. This does not apply for functions that receive a batch of messages.
+        /// </summary>
+        /// TODO the default value in Track 1 was the default value from the Track 1 SDK, which was 2000.
+        /// Verify that we are okay to diverge here.
+        public int MaxConcurrentSessions
+        {
+            get => _maxConcurrentSessions;
+
+            set
             {
-                batchOptions = new JObject
-                {
-                    { nameof(BatchOptions.MaxMessageCount), BatchOptions.MaxMessageCount },
-                    { nameof(BatchOptions.OperationTimeout), BatchOptions.OperationTimeout },
-                    { nameof(BatchOptions.AutoComplete), BatchOptions.AutoComplete },
-                };
+                Argument.AssertAtLeast(value, 1, nameof(MaxConcurrentSessions));
+                _maxConcurrentSessions = value;
             }
+        }
+        private int _maxConcurrentSessions = 8;
 
+        /// <summary>
+        /// Gets or sets an optional exception handler that will be invoked if an exception occurs while attempting to process
+        /// a message. This does not apply for functions that receive a batch of messages.
+        /// </summary>
+        public Func<ProcessErrorEventArgs, Task> ExceptionHandler { get; set; }
+
+        /// <summary>
+        /// Gets or sets an optional <see cref="TimeSpan"/> specifying the maximum time to wait when attempting to receive messages.
+        /// If not specified, the <see cref="ServiceBusRetryOptions.TryTimeout"/> will be used. This only applies for functions that receive
+        /// a batch of messages.
+        /// </summary>
+        /// TODO this should probably be exposed in the Processor as well and would then also apply for functions that receive a single message.
+        public TimeSpan? MaxWaitTime { get; set; }
```

Comment:
See https://github.com/Azure/azure-sdk-for-net/issues/16773 for discussion about exposing this. This is another use case.